back/src/captchaGen.ts

```diff
@@ -0,0 +1,158 @@
+import { genCaptchaString, getUid } from "./utils";
+import { createCanvas } from "canvas";
+
+import {
+  setCaptchaToken,
+  getCaptchaToken,
+  clearCaptchaToken
+} from "./common/redis/captcha";
+
+const CAPTCHA_LENGTH = 5;
+
+function randomInt(max) {
+  return Math.floor(Math.random() * max);
+}
+
+// CAPTCHA
+// IOT to provide captcha features we follow the following steps:
+// Captcha generation: captchaGen generates an image (data url) and a redis stored token associated to the captcha string
+// redis entries are short-lived (a few minutes) and deleted each time a checkCaptcha is performed (either success or failure)
+// Frontend displays the generated image, an input field, and stores the token value in an hidden field or react state entry
+// The input value and the token are sent back on form submit to be verified
+// Captcha verification: checkCaptcha verifies a given captcha input against the stored redis entry
+
+/**
+ * Store in redis a random uid:captchastring to be queried and verified
+ 
+ * @param captchaString
+ * @returns string
+ */
+
+export const captchaAsToken = async (
+  captchaString: string
+): Promise<string> => {
+  const token = getUid(10);
+
+  await setCaptchaToken(token, captchaString);
+  return token;
+};
+
+/**
+ * Check given input matches captcha string associated to a gievn captcha token
+ * @param inputCaptcha
+ * @param captchaToken
+ * @returns
+ */
+export async function checkCaptcha(captchaInput, captchaToken) {
+  if (!captchaInput || !captchaToken) {
+    return false;
+  }
+  // if token is expired, we get an empty string and the next equality fails
+  const captchaString = await getCaptchaToken(captchaToken);
+
+  const valid = !!captchaString && captchaString === captchaInput;
```

Comment:
Est-ce qu'on ne ferait pas une comparaison insensible à la casse ?
```suggestion
  const valid = !!captchaString && captchaString.toLowerCase() === captchaInput.toLowerCase();
```